backend/target_zone_estimation.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+import json
+import random
+from Step_Time_Calculation import calculate_step_time  # Ensure you have this imported
+
+def handle_threshold (threshold =20.0): 
+    return threshold
+    
+async def handle_data(websocket, threshold=None):
```

Comment:
function name seems largely vague. can we make it more descriptive? handle_data_streaming, handle_data_collection?